src/mdns/mdns_avahi.cpp

```diff
@@ -843,9 +767,38 @@ otbrError PublisherAvahi::TxtListToAvahiStringList(const TxtList &   aTxtList,
     return error;
 }
 
-std::string PublisherAvahi::MakeFullName(const std::string &aName)
+Publisher::ServiceRegistrationPtr PublisherAvahi::FindServiceRegistration(const AvahiEntryGroup *aEntryGroup) const
 {
-    return aName + "." + kDomain;
+    ServiceRegistrationPtr result = nullptr;
+
+    for (auto &kv : mServiceRegistrations)
+    {
+        auto &serviceReg = static_cast<AvahiServiceRegistration &>(*kv.second);
+        if (serviceReg.GetEntryGroup() == aEntryGroup)
+        {
+            result = kv.second;
+            break;
+        }
+    }
+
+    return result;
+}
+
+Publisher::HostRegistrationPtr PublisherAvahi::FindHostRegistration(const AvahiEntryGroup *aEntryGroup) const
+{
+    HostRegistrationPtr result = nullptr;
+
+    for (auto &kv : mHostRegistrations)
+    {
+        auto &hostReg = static_cast<AvahiHostRegistration &>(*kv.second);
```

Comment:
```suggestion
    for (const auto &kv : mHostRegistrations)
    {
        const auto &hostReg = static_cast<const AvahiHostRegistration &>(*kv.second);
```
nit. Feel free to leave it as is.